pkg/controllermanager/controller/managedseedset/managedseedset_actuator.go

```diff
@@ -0,0 +1,414 @@
+		a.infoEventf(set, "Waiting for shoot %s to be deleted", r.GetFullName())
+		updatePendingReplica(status, r.GetName(), seedmanagementv1alpha1.ShootDeletingReason, nil)
+		return true, nil
+
+	case replicaStatus == StatusShootReconciled:
+		// This replica's shoot is fully reconciled and its managed seed doesn't exist
+		// If not scaling in, create its managed seed, otherwise delete its shoot
+		if !scalingIn {
+			a.infoEventf(set, "Creating managed seed %s", r.GetFullName())
+			if err := r.CreateManagedSeed(ctx, a.gardenClient.Client()); err != nil {
+				return false, err
+			}
+			updatePendingReplica(status, r.GetName(), seedmanagementv1alpha1.ManagedSeedPreparingReason, nil)
+		} else {
+			a.infoEventf(set, "Deleting shoot %s", r.GetFullName())
+			if err := r.DeleteShoot(ctx, a.gardenClient.Client()); err != nil {
+				return false, err
+			}
+			updatePendingReplica(status, r.GetName(), seedmanagementv1alpha1.ShootDeletingReason, nil)
+		}
+		return true, nil
+
+	case replicaStatus == StatusManagedSeedPreparing && !scalingIn:
+		// This replica's managed seed is preparing, wait for the it to be registered before moving to the next replica
+		a.infoEventf(set, "Waiting for managed seed %s to be registered", r.GetFullName())
+		updatePendingReplica(status, r.GetName(), seedmanagementv1alpha1.ManagedSeedPreparingReason, nil)
+		return true, nil
+
+	case replicaStatus == StatusManagedSeedDeleting:
+		// This replica's managed seed is deleting, wait for it to be deleted before moving to the next replica
+		a.infoEventf(set, "Waiting for managed seed %s to be deleted", r.GetFullName())
+		updatePendingReplica(status, r.GetName(), seedmanagementv1alpha1.ManagedSeedDeletingReason, nil)
+		return true, nil
+
+	case !r.IsSeedReady() && !scalingIn:
+		// This replica's seed is not ready, wait for it to be ready before moving to the next replica
+		a.infoEventf(set, "Waiting for seed %s to be ready", r.GetName())
+		updatePendingReplica(status, r.GetName(), seedmanagementv1alpha1.SeedNotReadyReason, nil)
+		return true, nil
+
+	case r.GetShootHealthStatus() != operationshoot.StatusHealthy && !scalingIn:
+		// This replica's shoot is not healthy, wait for it to be healthy before moving to the next replica
+		a.infoEventf(set, "Waiting for shoot %s to be healthy", r.GetFullName())
+		updatePendingReplica(status, r.GetName(), seedmanagementv1alpha1.ShootNotHealthyReason, nil)
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func (a *actuator) createReplica(
+	ctx context.Context,
+	set *seedmanagementv1alpha1.ManagedSeedSet,
+	status *seedmanagementv1alpha1.ManagedSeedSetStatus,
+	ordinal int,
+) error {
+	r := a.replicaFactory.NewReplica(set, nil, nil, nil, false)
+	a.infoEventf(set, "Creating shoot %s", getFullName(set, ordinal))
+	if err := r.CreateShoot(ctx, a.gardenClient.Client(), ordinal); err != nil {
+		return err
+	}
+	updatePendingReplica(status, r.GetName(), seedmanagementv1alpha1.ShootReconcilingReason, nil)
```

Comment:
We should set the pending replica before creating a new replica (same applies to the deletion below).
If a Shoot fails to be created because of some reason, the controller will post an event, but the pending replica is not set (which is important for further reconciliations), e.g.:
```
3m18s       Warning   ReconcileError   managedseedset/gcp-europe-west1   Could not reconcile creation or update: the length of the shoot name and the project name must not exceed 21 characters (project: garden; shoot: gcp-europe-west1-0)
```

Reply:
If the shoot fails to be created, this will result in an error, which will be automatically requeued with backoff. The same applies to all errors elsewhere in this method. It doesn't matter that there is no pending replica, the reconcile will be retried until the error is gone, and at that point the pending replica will be set appropriately. I don't think anything should be changed.

Reply:
I tested this, it behaves exactly as I expected it should. I got multiple identical errors from the multiple failed attempts to reconcile the MSS:

```
Warning  ReconcileError  84s (x12 over 106s)  gardener-controller-manager  Could not reconcile creation or update: the length of the shoot name and the project name must not exceed 21 characters (project: garden; shoot: i500152-gcpx-long-name-should-fail-0)
```

I don't think we should do anything more here.